plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/tasks/BaseKtLintCheckTask.kt

```diff
@@ -0,0 +1,152 @@
+import org.jlleitschuh.gradle.ktlint.applyGitFilter
+import org.jlleitschuh.gradle.ktlint.getEditorConfigFiles
+import org.jlleitschuh.gradle.ktlint.intermediateResultsBuildDir
+import org.jlleitschuh.gradle.ktlint.property
+import org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction
+import java.io.File
+import java.util.concurrent.Callable
+import javax.inject.Inject
+
+@Suppress("UnstableApiUsage")
+abstract class BaseKtLintCheckTask @Inject constructor(
+    objectFactory: ObjectFactory,
+    projectLayout: ProjectLayout,
+    private val workerExecutor: WorkerExecutor,
+) : SourceTask() {
+
+    @get:Classpath
+    internal abstract val ktLintClasspath: ConfigurableFileCollection
+
+    @get:Internal
+    internal abstract val additionalEditorconfigFile: RegularFileProperty
+
+    @Suppress("unused")
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    internal val editorConfigFiles: FileCollection by lazy(LazyThreadSafetyMode.NONE) {
+        // Gradle will lazy evaluate this task input only on task execution
+        getEditorConfigFiles(project, additionalEditorconfigFile)
+    }
+
+    @get:Input
+    internal abstract val ktLintVersion: Property<String>
+
+    @get:Classpath
+    internal abstract val ruleSetsClasspath: ConfigurableFileCollection
+
+    @get:Input
+    internal abstract val debug: Property<Boolean>
+
+    @get:Input
+    internal abstract val android: Property<Boolean>
+
+    @get:Input
+    internal abstract val disabledRules: SetProperty<String>
+
+    @get:Input
+    internal abstract val enableExperimentalRules: Property<Boolean>
+
+    /**
+     * Max lint worker heap size. Default is "256m".
+     */
+    @get:Internal
+    val workerMaxHeapSize: Property<String> = objectFactory.property {
+        convention("256m")
+    }
+
+    init {
+        if (project.hasProperty(FILTER_INCLUDE_PROPERTY_NAME)) {
+            applyGitFilter()
+        } else {
+            KOTLIN_EXTENSIONS.forEach {
+                include("**/*.$it")
+            }
+        }
+    }
+
+    @ReplacedBy("stableSources")
+    override fun getSource(): FileTree {
+        return super.getSource()
+    }
+
+    @get:SkipWhenEmpty
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    internal val stableSources: FileCollection = project.files(
+        Callable {
+            return@Callable source
+        }
+    )
+
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFile
+    internal abstract val loadedReporters: RegularFileProperty
+
+    @get:OutputFile
+    internal val discoveredErrors: RegularFileProperty = objectFactory
+        .fileProperty()
+        .convention(
+            projectLayout.intermediateResultsBuildDir("${name}_errors.bin")
+        )
+
+    protected fun runLint(
+        filesToCheck: Set<File>,
+        formatSources: Boolean,
+    ) {
+        checkDisabledRulesSupportedKtLintVersion()
+
+        val queue = workerExecutor.processIsolation { spec ->
```

Comment:
Can you document why you're using process isolation here vs classpath isolation?

Reply:
Fixed via 8d561a1